core/os/os2/process.odin

```diff
@@ -3,100 +3,399 @@ package os2
 import "core:sync"
 import "core:time"
 import "base:runtime"
+import "core:strings"
 
-args: []string
+/*
+	In procedures that explicitly state this as one of the allowed values,
+	specifies an infinite timeout.
+*/
+TIMEOUT_INFINITE :: time.MIN_DURATION // Note(flysand): Any negative duration will be treated as infinity
 
+/*
+	Arguments to the current process.
+
+	See `get_args()` for description of the slice.
+*/
+args := get_args()
+
+/*
+	Obtain the process argument array from the OS.
+
+	Slice, containing arguments to the current process. Each element of the
+	slice contains a single argument. The first element of the slice would
+	typically is the path to the currently running executable.
+*/
+get_args :: proc() -> []string {
+	result := make([]string, len(runtime.args__), allocator = context.allocator)
```

Comment:
This should use `heap_allocator()` not `context.allocator`, and also be a private call.